Tests/AsyncAlgorithmsTests/TestZipLatestFrom.swift

```diff
@@ -0,0 +1,782 @@
+
+    await other.send("a")
+
+    do {
+      let value = try await iterator.next()
+      XCTFail("got \(value as Any) but expected throw")
+    } catch {
+      XCTAssertEqual(error as? Failure, Failure())
+    }
+  }
+
+  func test_zipLatestFrom_throws_when_other_throws() async {
+    let base = Indefinite(value: 1)
+    let other = AsyncThrowingChannel<String, Error>()
+    let sequence = base.async.zipLatest(from: other)
+    var iterator = sequence.makeAsyncIterator()
+
+    await other.fail(Failure())
+
+    do {
+      var element: (Int, String)?
+      repeat {
+        element = try await iterator.next()
+      } while element == nil
+      XCTFail("got \(element as Any) but expected throw")
+    } catch {
+      XCTAssertEqual(error as? Failure, Failure())
+    }
+  }
+
+  func test_zipLatestFrom_uses_latest_element_from_other_when_other_produces_first_elements() async {
+    // Timeline
+    // base:     |---1------2----------3-------
+    // other:    |-a-|------|-----b-c--|-------
+    // expected: |--(1, a)-(2, a)-----(3, c)---
+    
+    let otherHasProducedA = expectation(description: "Other has produced 'a'")
+    let otherHasProducedC = expectation(description: "Other has produced 'c'")
+    let finished = expectation(description: "finished")
+
+    let base = AsyncChannel<Int>()
+    let other = AsyncChannel<String>()
+    let spyOther = AsyncSpySequence(other, onLatestKnownElement: { latestKnownElement in
+      if latestKnownElement == "a" {
+        otherHasProducedA.fulfill()
+      }
+      if latestKnownElement == "c" {
+        otherHasProducedC.fulfill()
+      }
+    })
+
+    let sequence = base.zipLatest(from: spyOther)
+
+    let validator = Validator<(Int, String)>()
+    validator.test(sequence) { iterator in
+      let pastEnd = await iterator.next()
+      XCTAssertNil(pastEnd)
+      finished.fulfill()
+    }
+
+    var value = await validator.validate()
+    XCTAssertEqual(value, [])
+
+    await other.send("a")
+    value = validator.current
+    XCTAssertEqual(value, [])
+
+    wait(for: [otherHasProducedA], timeout: 1)
+
+    await base.send(1)
+    value = await validator.validate()
+    XCTAssertEqual(value, [(1, "a")])
+
+    await base.send(2)
+    value = await validator.validate()
+    XCTAssertEqual(value, [(1, "a"), (2, "a")])
+
+    await other.send("b")
+    await other.send("c")
+    value = validator.current
+    XCTAssertEqual(value, [(1, "a"), (2, "a")])
+
+    wait(for: [otherHasProducedC], timeout: 1)
+
+    await base.send(3)
+    value = await validator.validate()
+    XCTAssertEqual(value, [(1, "a"), (2, "a"), (3, "c")])
+
+    await base.finish()
+
+    wait(for: [finished], timeout: 1.0)
+    value = validator.current
+    XCTAssertEqual(value, [(1, "a"), (2, "a"), (3, "c")])
+  }
+
+  func test_zipLatestFrom_uses_latest_element_from_other_when_base_produces_first_elements() async {
+    // Timeline
+    // base:     |-1-2---3------4---------5-----
```

Comment:
these diagrams are really close to the validation diagram, however the use of `|` is kinda confusing.

Perhaps we can use some sort of combing function that is easily understandable to represent the tuple? e.g. we can use emoji so a tuple of a shape out of 🤍⬜️⚪️ along side a color out of 🟥🟧🟨🟩🟦🟪⬛️ results in a symbol of ❤️🟥🔴🧡🟧🟠... etc?

Doing a 3 variant might be hard; I still need to think on how to accomplish that.